pkg/detector/detector.go

```diff
@@ -560,6 +560,7 @@ func (d *ResourceDetector) ApplyClusterPolicy(object *unstructured.Unstructured,
 				// Just update necessary fields, especially avoid modifying Spec.Clusters which is scheduling result, if already exists.
 				bindingCopy.Annotations = util.DedupeAndMergeAnnotations(bindingCopy.Annotations, binding.Annotations)
 				bindingCopy.Labels = util.DedupeAndMergeLabels(bindingCopy.Labels, binding.Labels)
+				bindingCopy.Finalizers = util.DedupeAndMergeFinalizers(bindingCopy.Finalizers, binding.Finalizers)
```

Comment:
Is this the only place that needs modification?
My understanding is that after the detector creates ResourceBindng/ClusterResourceBinding, downstream controllers could potentially append finalizers. Therefore, I think all places where updating RB/CRB should be fixed.

@XiShanYongYe-Chang What's your opinion?

Reply:
>  I think all places where updating RB/CRB should be fixed

Oh

Reply:
One more place:
https://github.com/karmada-io/karmada/blob/3ae9f5e99f3153ff52f90a63af004929a51ae63f/pkg/detector/detector.go#L470-L469

Reply:
done

Reply:
For the CRB, it is OK not to modify the CRB, because the dependency distribution does not add the related finalizer to the CRB resource. 

Of course, the modification is better, which can avoid the same problem in the future.